test/test_linear_algebra.jl

```diff
@@ -94,4 +94,37 @@ fs =  parsed_args["fs"]
         @test isapprox(norm(w_out.weights[1][:, 1:19]), 0f0; rtol=ftol)
         mul!(w_out, M', w)
         @test isapprox(w_out, M'*w; rtol=ftol)
+
+        # test in-place
+
+        dobs1 = deepcopy(dobs)
+        dobs2 = deepcopy(dobs)
+        dobs3 = deepcopy(dobs)
+        dobs4 = deepcopy(dobs)
+
+        temp1 = F*dobs
+        temp2 = F'*dobs
+        temp3 = Md*dobs
+        temp4 = Md'*dobs
```

Comment:
that's not very efficient testing, it's only testing half of the issue. Also no idea what all these dobs2,3,4 are they don't serve any purpose.

```
dobs1 = deepcopy(dobs)
for Op in [F, F', Md , Md']
    m = Op*dobs 
     # Test that dobs wasn't modified
    @test dobs == dobs1
     # Test that it did compute something 
      @test m ~= dobs
```

same below for w

Reply:
Modified both dobs and w in the latest commit